trie/branchNode.go

```diff
@@ -653,36 +717,74 @@ func (bn *branchNode) delete(data []core.TrieData, db common.TrieStorageInteract
 
 	numChildren, pos := getChildPosition(bn)
 	if numChildren == 0 {
-		return true, nil, modifiedHashes, nil
+		return true, nil, modifiedHashes
 	}
 	if numChildren == 1 {
 		err = resolveIfCollapsed(bn, byte(pos), db)
 		if err != nil {
-			return false, nil, emptyHashes, err
+			goRoutinesManager.SetError(err)
+			return false, nil, [][]byte{}
 		}
 
 		err = resolveIfCollapsed(bn.children[pos], byte(pos), db)
 		if err != nil {
-			return false, nil, emptyHashes, err
+			goRoutinesManager.SetError(err)
+			return false, nil, [][]byte{}
 		}
 
 		var newChildHash bool
 		newNode, newChildHash, err := bn.children[pos].reduceNode(pos)
 		if err != nil {
-			return false, nil, emptyHashes, err
+			goRoutinesManager.SetError(err)
+			return false, nil, [][]byte{}
 		}
 
 		if newChildHash && !bn.children[pos].isDirty() {
 			modifiedHashes = append(modifiedHashes, bn.children[pos].getHash())
 		}
 
-		return true, newNode, modifiedHashes, nil
+		return true, newNode, modifiedHashes
 	}
 
-	return true, bn, modifiedHashes, nil
+	return true, bn, modifiedHashes
+}
+
+func (bn *branchNode) deleteChild(
+	dataForRemoval []core.TrieData,
+	childPos int,
+	hasBeenModified *atomic.Flag,
```

Comment:
return `hasBeenModified` as a variable similar to `insertOnChild` and set atomic flag as in `insert` case? this way the dataForInsertion(Removal) flows are very similar, they might be able to be extracted in a common function  or component since `insetOnChild` and `deleteChild` would have the same singature

Reply:
Refactored.